src/poetry/core/version/markers.py

```diff
@@ -357,7 +356,10 @@ def __init__(
                 values = re.split("[ ,]+", self._value)
                 constraint_string = glue.join(f"{op} {value}" for value in values)
 
+        try:
             parsed_constraint = parser(constraint_string)
+        except ParseConstraintError as e:
+            raise InvalidMarker(f"Invalid marker: {original_constraint_string}") from e
```

Comment:
Since it's only the constraint maybe as follows?

```suggestion
            raise InvalidMarker(f"Invalid marker constraint for "{name}": {original_constraint_string}") from e
```